mdagent/tools/base_tools/analysis_tools/secondary_structure.py

```diff
@@ -0,0 +1,301 @@
+
+
+def plot_x_over_time(x, values, traj_id, path_registry):
+    plt.figure(figsize=(10, 6))
+    plt.plot(values)
+    plt.xlabel("Frame")
+    plt.ylabel(x)
+    plt.title(f"{x} Over Time")
+    plt.grid(True)
+
+    file_name = path_registry.write_file_name(
+        FileType.FIGURE,
+        file_format="png",
+    )
+    file_id = path_registry.get_fileid(file_name, FileType.RECORD)
+
+    file_path = f"{path_registry.ckpt_figures}/{x}_over_time_{traj_id}.png"
+    plt.savefig(file_path, format="png", dpi=300, bbox_inches="tight")
+    plt.close()
+
+    path_registry.map_path(
+        file_id,
+        file_name,
+        description=(f"{x} plot for trajectory " f"with id: {traj_id}"),
+    )
+    return file_id
+
+
+class ComputeAsphericity(BaseTool):
+    name = "ComputeAsphericity"
+    description = """Compute the asphericity for each frame in a
+    molecular dynamics trajectory. Input is a trajectory file (e.g., .
+    xtc, .trr) and an optional topology file (e.g., .pdb, .prmtop).
+    The output is asphericity values for each frame of the
+    trajectory."""
+    path_registry: PathRegistry | None = None
+
+    def __init__(self, path_registry: PathRegistry):
+        super().__init__()
+        self.path_registry = path_registry
+
+    def _compute_asphericity(self, traj):
+        return md.asphericity(traj)
+
+    def _run(self, traj_file, top_file=None):
+        try:
+            traj = load_single_traj(
+                path_registry=self.path_registry,
+                traj_fileid=traj_file,
+                top_fileid=top_file,
+            )
+            if not traj:
+                raise Exception("Trajectory could not be loaded.")
+        except Exception as e:
+            return str(e)
+        asphericity_values = self._compute_asphericity(traj)
+        raw_file_id = write_raw_x(
+            "asphericity", asphericity_values, traj_file, self.path_registry
+        )
+        plot_file_id = plot_x_over_time(
+            "Asphericity", asphericity_values, traj_file, self.path_registry
+        )
+        return (
+            "asphericity_values saved to "
+            f"{raw_file_id}, plot saved to "
+            f"{plot_file_id}"
+        )
+
+    async def _arun(self, traj_file, top_file=None):
+        raise NotImplementedError("Async version not implemented")
+
+
+class ComputeAcylindricity(BaseTool):
+    name = "ComputeAcylindricity"
+    description = """Compute the acylindricity for each frame in a
+    molecular dynamics trajectory. Input is a trajectory file (e.g., .
+    xtc, .trr) and an optional topology file (e.g., .pdb, .prmtop). The
+    output is an array of acylindricity values for each frame of the
+    trajectory."""
+    path_registry: PathRegistry | None = None
+
+    def __init__(self, path_registry: PathRegistry):
+        super().__init__()
+        self.path_registry = path_registry
+
+    def _compute_acylindricity(self, traj):
```

Comment:
It would be nice to include a little doc string on what is the output of the function.

Reply:
the same in the other tools